llvm/include/llvm/IR/VectorUtils.h

```diff
@@ -0,0 +1,74 @@
+//===----------- VectorUtils.h - Vector type utility functions -*- C++ -*-====//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_IR_VECTORUTILS_H
+#define LLVM_IR_VECTORUTILS_H
+
+#include "llvm/IR/DerivedTypes.h"
+#include "llvm/IR/StructWideningUtils.h"
```

Comment:
Is there a benefit from having a separate `StructWideningUtils.h` instead of inlining it here?

Reply:
To be fair to @MacDue, it was my suggestion. I was thinking that the separation between widening struct types and scalar types was useful to have and perhaps a bit clearer, especially if at some point the logic for structs becomes more complex. If you think it's better to do everything within just VectorUtils.h and VectorUtils.cpp that's also fine. @MacDue what do you think?

Reply:
I don't have have much of a preference here, I think I originally just placed everything in `VectorUtils.{h|cpp}`, before splitting the structs to `CallWideningUtils.h` (now `StructWideningUtils.h`). 

As long as everyone is okay with the final placement I'm okay with either way :slightly_smiling_face:

Reply:
IMO it would be good to keep things together in a single header given they are both quite small at the moment and I don't think we anticipate any users including just `StructWideningUtils.h` but not `VectorUtils.h`

Reply:
Moved all the helpers into `VectorUtils.{h|cpp}` :+1: